### PR TITLE
Decoder improvements

### PR DESCRIPTION
**Pull request by oschwald**
These either simplify the code or slightly improve performance.

### CODE REVIEW

File: maxminddb/decoder.py [hunk 1]
```diff
@@ -164,21 +164,22 @@ def _verify_size(self, expected, actual):
...
-        if type_num == 1:
+        if type_num == 1 or size < 29:
             return size, offset
-        bytes_to_read = 0 if size < 29 else size - 28
-
-        new_offset = offset + bytes_to_read
-        size_bytes = self._buffer[offset:new_offset]
 
         # Using unpack rather than int_from_bytes as it is about 200 lookups
         # per second faster here.
```
Comment (horgh) [inline maxminddb/decoder.py]: Should we update this comment?

File: maxminddb/file.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def __getitem__(self, key):
         if isinstance(key, slice):
             return self._read(key.stop - key.start, key.start)
         if isinstance(key, int):
-            return self._read(1, key)
+            return self._read(1, key)[0]
```
Comment (horgh) [inline maxminddb/file.py]: What was the difference?

Comment (oschwald) [reply maxminddb/file.py]: In Python 3, the buffers are `bytes`, which is a sequence of bytes. When you index a `bytes`, you get back an integer in the range 0 to 255, but the file read was returning a `bytes` of length 1. This different did not matter previously, but with the other changes, it did.

File: maxminddb/decoder.py (hunk 1 above)
Comment (oschwald) [reply maxminddb/decoder.py]: e850475

**Pull request merged**